controllers/telemetry/collector_controller.go

```diff
@@ -143,19 +131,38 @@ func (r *CollectorReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *CollectorReconciler) updateStatus(ctx context.Context, obj client.Object) error {
-	return r.Status().Update(ctx, obj)
+// handleTenantReconcileError handles errors that occur during reconciliation steps
+func (r *CollectorReconciler) handleCollectorReconcileError(ctx context.Context, baseManager *manager.BaseManager, collector *v1alpha1.Collector, err error) (ctrl.Result, error) {
+	switch {
+	case errors.Is(err, manager.ErrTenantFailed): // This error indicates that the tenant is in a failed state, and we should requeue after a delay.
+		return ctrl.Result{RequeueAfter: requeueDelayOnFailedTenant}, err
```

Comment:
I beleive if we return with an error the requeue doesn't count, but it will result in a similar backoff based retry